### PR TITLE
fix memory leak issue

### PR DESCRIPTION
**Pull request by dawnbreaks**
release inactive long-polling channel from ClientsBox to avoid memory leak

### CODE REVIEW

**Comment by dawnbreaks** [comment]
Hi,  there is memory leak issue in our prod environment. We analyzed the heap dump, and found that the  ClientsBox consumed huge memory.  Inactive Channels had not be released from ClientsBox.

**Comment by mrniko** [comment]
Would you review my comments?

**Comment by dawnbreaks** [comment]
Where 's your comments?

**Comment by mrniko** [comment]
https://github.com/mrniko/netty-socketio/pull/572/commits/183bdebda86d6fe2d55927d2edf86586d55069a5

File: src/main/java/com/corundumstudio/socketio/transport/PollingTransport.java [hunk 1]
```diff
@@ -186,4 +179,16 @@ private void sendError(ChannelHandlerContext ctx) {
...
 
+    @Override
+    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
+        final Channel channel = ctx.channel();
+        ClientHead client = clientsBox.get(channel);
+        if (client != null && client.isTransportChannel(ctx.channel(), Transport.POLLING)) {
+            log.debug("channel inactive {}", client.getSessionId());
+            client.onChannelDisconnect();
+        }
+        channel.close();
```
Comment (mrniko) [inline src/main/java/com/corundumstudio/socketio/transport/PollingTransport.java]: @dawnbreaks You don't need to invoke channel close() method.

**Comment by dawnbreaks** [comment]
Hi, please review the code again, thanks very much.

**Comment by mrniko** [comment]
Thank you!

**Pull request merged**